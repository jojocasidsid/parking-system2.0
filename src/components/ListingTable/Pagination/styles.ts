import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
