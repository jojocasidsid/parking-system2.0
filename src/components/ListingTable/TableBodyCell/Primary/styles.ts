import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
`
