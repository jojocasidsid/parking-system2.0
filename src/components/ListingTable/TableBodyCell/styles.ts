import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'
import TableRow from '@mui/material/TableRow'

export const StyledTableRow = styled(TableRow)`
	cursor: pointer;
`
export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
`
