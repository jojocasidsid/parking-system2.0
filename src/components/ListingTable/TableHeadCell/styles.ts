import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTableHeadTypography = styled(Typography)`
	font-weight: bold;
	font-size: 14px;
`
