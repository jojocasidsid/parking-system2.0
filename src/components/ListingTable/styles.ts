import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
