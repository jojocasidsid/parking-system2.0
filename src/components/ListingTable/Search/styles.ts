import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
