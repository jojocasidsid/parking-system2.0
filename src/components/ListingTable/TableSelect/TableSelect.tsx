import React from 'react'
import { MenuItem } from '@mui/material'
import { StyledRoot, StyledLabel, StyledSelect } from './styles'
import { IFilters, IOptions } from '../types'

interface IProps {
	filters?: IFilters
	setFilters?: React.Dispatch<React.SetStateAction<IFilters>>
	label?: string
	name: keyof IFilters
	fieldOptions?: IOptions[]
}

const TableSelect = ({
	filters,
	setFilters,
	name,
	label,
	fieldOptions,
}: IProps) => (
	<StyledRoot fullWidth={false}>
		{label && <StyledLabel htmlFor={name}>{label}</StyledLabel>}
		<StyledSelect
			name={name}
			id={name}
			value={filters && filters[name]}
			onChange={(event) =>
				setFilters &&
				setFilters((prev) => ({
					...prev,
					[name]: event.target.value as string,
				}))
			}
			// eslint-disable-next-line react/no-unstable-nested-components
		>
			{fieldOptions?.map((row, index) => (
				<MenuItem value={row.value} key={`status-${index}-${row.value}`}>
					{row.label}
				</MenuItem>
			))}
		</StyledSelect>
	</StyledRoot>
)

export default TableSelect
