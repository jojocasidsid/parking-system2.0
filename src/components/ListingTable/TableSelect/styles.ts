import styled from '@emotion/styled'
import { Theme } from '@mui/material/styles'

import { Select } from '@mui/material'

interface rootInterface {
	fullWidth?: boolean
}
interface IStyled {
	theme?: Theme
}
interface IStyledSelect extends IStyled {
	pagination?: boolean
}

export const StyledRoot = styled('div')`
	${(props: rootInterface) =>
		props.fullWidth ? 'width: 100%' : 'max-width: 225px'};
	display: block;
`

export const StyledLabel = styled('label')`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;
	font-style: normal;
	font-size: 14px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	color: #303643;
	opacity: 0.5;

	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledSelect = styled(Select)`
	min-width: ${({ pagination }: IStyledSelect) =>
		pagination ? '60px' : '150px'};

	& .MuiSelect-select {
		padding: 9px 24px 7px 12px !important;
		font-style: normal;
		font-size: 12px;
		line-height: 14px;
		font-weight: bold;
	}
	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	&.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}
`
