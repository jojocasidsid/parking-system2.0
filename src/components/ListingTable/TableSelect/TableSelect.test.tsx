import React, { useState } from 'react'
import renderer from 'react-test-renderer'

// packages
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

// types
import { IFilters } from 'hooks/useListingTable'

// component
import TableSelect from './TableSelect'

const SHOW_ITEMS_OPTIONS = [
	{ value: 10, label: '10' },
	{ value: 50, label: '50' },
	{ value: 100, label: '100' },
	{ value: 1000, label: '1000' },
]

describe('Table Selext Component', () => {
	it('should render Table select component without crashing', () => {
		const Component = () => {
			const [filters, setFilters] = useState<IFilters>({
				_order: 'asc',
				_sort: '',
				_page: 1,
				_limit: 20,
				q: '',
			})
			return (
				<ThemeProvider theme={theme}>
					<TableSelect
						name='_limit'
						filters={filters}
						setFilters={setFilters}
						fieldOptions={SHOW_ITEMS_OPTIONS}
					/>
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)
		expect(tree).toMatchSnapshot()
	})
})
