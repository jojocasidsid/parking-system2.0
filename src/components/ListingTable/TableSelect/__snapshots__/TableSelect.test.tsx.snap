// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Selext Component should render Table select component without crashing 1`] = `
<div
  className="css-dwoubo"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-1kjct6a-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="_limit"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-jcfq8n-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="_limit"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      name="_limit"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={20}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
